### ISSUE TITLE
Support for markdown-it plugins and modifying react-markdown

### ISSUE THREAD

**Issue by ortonomy** [OpusCapita/react-markdown#144]
We'd like to use react-markdown in a project, but we'd like to extend it to support markdown-it plugins so and extended markdown features

Do you have a guide on adding buttons and extending the markdown that this awesome component can support can support?

**Comment by kvolkovich-sc** [comment]
Hi, @ortonomy.

We currently don't have guide on adding buttons and decorators.

You can by and [slatejs documentation](https://docs.slatejs.org/) https://github.com/OpusCapita/react-markdown/blob/b060286bbd8c8426636c8b48ee4b483bb3b85f81/package.json#L100
and by diving into **react-markdown** source code.

I'm not sure if this is a very simple task.

Also I'm not sure that you need text centering and underline into markdown editor.

Of course you can do it by using html tags into markdown. But you must understand that this editor is raw markdown text editor (at the current time).

**Comment by ortonomy** [comment]
@kvolkovich-sc -- thanks for answering so fast.

Perhaps  I can let you know about our requirements:

1. We will have editors who will be creating content such that they require to be able to mark pieces of text as formatted (an exhaustive list): bold, italic, underline, highlighted, centered
2. We don't want to store HTML output in our DB, we want to store it in markdown (it's more like meta data that way, not the resulting presentation)
3. We then want to show a preview to the user after, or during
4. We will want to extend the metadata with our domain markdown later (for more complicated formats: e.g mathjax, custom formats). 

``react-markdown`` uses ``markdown-it`` to parse the markdown -- which supports plugins for mathjax, but also custom markdown decorators, react-markdown seems like an ideal candidate for creating our own custom decorators and making them parse 

> But you must understand that this editor is raw markdown text editor (at the current time) 

I get this, but ...

React-markdown seems to format the markdown to show the resulting format e.g., bold, italic, strikethrough e.g.:

![image](https://user-images.githubusercontent.com/6688676/44777000-0c118700-abac-11e8-9413-6caf28fe555c.png)

So basically, we'd like to be able to add decorators, (e.g. $$ 1^2 $$ for mathjax) and add them as a button too, and then see the result of the markdown, either in separate preview, or partially formatted inside the editor. 

Does that make sense?

**Comment by kvolkovich-sc** [comment]
Yes, it make sense.

You can add additional buttons which can insert custom text right now. Its not complex.

But I'm not sure about visual decorators by adding CSS class or something like that.

Sure, it should be possible by making a little code changes.

Right now we have no resources ⏳ to support react-markdown out of business cases bounds.

If you want, you can dive into codebase and make a Pull Request.
